components/sync/engine/brave_syncer_device_poll_unittest.cc

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/sync/engine/brave_syncer_device_poll.h"
+
+#include <memory>
+
+#include "base/functional/callback.h"
+#include "base/time/time_override.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+using base::subtle::ScopedTimeClockOverrides;
+using base::subtle::TimeTicksNowIgnoringOverride;
+
+namespace syncer {
+
+namespace {
+
+base::TimeDelta g_delta_ticks_overrides;
+std::unique_ptr<ScopedTimeClockOverrides> OverrideTicks() {
+  return std::make_unique<ScopedTimeClockOverrides>(
```

Comment:
we can just use `base::ScopedMockClockOverride`. it does exactly what you're implementing here manually :)

Reply:
wow, thanks, I wish I would see this class before :+1: 
fixed at https://github.com/brave/brave-core/pull/17077/commits/1641b453a9ed2a2f4439d8b13b6112121c0ca4de#diff-675d7546395450763f3effe1ebfd03a4949495f150109b0928b91f87f249895dR26